### PR TITLE
Merge openjdk jdk8u302-b01

### PR DESCRIPTION
**Pull request by keithc-ca**
Conflicts:
```
  jdk/make/Images.gmk
  make/common/NativeCompilation.gmk
```

### CODE REVIEW

File: jdk/make/Images.gmk
Comment (JasonFengJ9) [inline jdk/make/Images.gmk:178]: Is removing these `macosx` JCL patch intentional?

Comment (keithc-ca) [reply jdk/make/Images.gmk:178]: Yes, those files will still be removed by the [build script](https://github.com/adoptium/temurin-build/blob/master/sbin/build.sh#L770).

**Comment by JasonFengJ9** [comment]
There are lots of `Merge latest openjdk level jdk8u222-b01` alike commits which usually are not observed in past merging PRs.
Does this PR creation follow the procedure specified in merging script job output?
```
23:03:56  1. cd merge_tmp
23:03:56  2. Resolve conflict... and commit
23:03:56  3. Update OPENJDK_TAG level...
23:03:56       git tag --merged openj9-staging jdk8u*-b* | grep -v _openj9 | cut -c6- | awk -F'[\-b]+' '{print $1"-"$2}' | sort -t- -k2n | sort -t- -k1n -s | sed 's/^/jdk8u/' | sed 's/-/-b/' | tail -1
23:03:56           <BUILD_TAG>
23:03:56       echo 'OPENJDK_TAG := <BUILD_TAG>' > closed/openjdk-tag.gmk
23:03:56       git add closed/openjdk-tag.gmk
23:03:56       git commit -s -m'Update OPENJDK_TAG to merged level <BUILD_TAG>'
23:03:56  4. git merge -m'Merging latest openj9 into openj9-staging' origin/openj9
23:03:56  5. git push --tags -u origin openj9-staging
```

**Comment by keithc-ca** [comment]
Yes, I followed the same procedure, but with more consistent commit messages. I think that script should be updated likewise.

**Comment by JasonFengJ9** [comment]
The commit message wasn't the concern, it is the number of earlier merging commits.
Earlier merge commits

**Comment by keithc-ca** [comment]
I think I understand the reason for the difference. I created a new branch from 'openj9' (which is ahead of 'openj9-staging' so that should not matter) and then worked to merge from 'master' (at d9f8d65fa883) instead of from 62c70425e09a like the script would do. The result should be the same because d9f8d65fa883 and 62c70425e09a refer to identical content, even though all those merge commits will be ancestors of the openj9-staging branch once this is merged.

That's one of the features in the merge script I'd like to fix: it need not create the extra commit (d9f8d65fa883 in this instance): master could refer to 62c70425e09a instead.

**Comment by JasonFengJ9** [comment]
> I created a new branch from 'openj9' (which is ahead of 'openj9-staging' so that should not matter) and then worked to merge from 'master' (at d9f8d65)

This seems bring those earlier unrelated commits into this PR.
I think this PR should only contain the new commits since last update, the involvement of earlier commits (such as dated as Mar 6, 2014) is not desirable.

**Comment by keithc-ca** [comment]
If the script worked as it should, all those commits would already be ancestors of the openj9-staging branch. Doing as you proposed yields identical content. Those merges are part of the history of the project (and are already ancestors of the openj9 branch).

**Comment by JasonFengJ9** [comment]
This PR has same contents as a usual PR, and probably eventually yields same result in the repo as you predicted.
On the other hand, those earlier and unrelated commits in the PR complicated the review and might introduce confusion for future references, does this warrant a PR with only new updates?

**Comment by keithc-ca** [comment]
It's not 'eventually', it's now. But, on reflection, I agree the merge commits can wait until the merge script is fixed: I'll update this as requested.

**Comment by keithc-ca** [comment]
Notice:
> 5a7bfad78c02c80c83b9f80413e5d84f6342ad32 and 92da30f685cacf1bbc608124bad9bc97a8a1ad15 are identical.

**Comment by JasonFengJ9** [comment]
Can you remove those commits dated year 2014/../2020?

**Comment by keithc-ca** [comment]
I'm not sure that's possible. For example, 9293f899115e104ae1b7fb8e87504fc61c9a7f74 (a commit from Mar 6, 2014) is an ancestor of 62c70425e09a7299c7ca85694d95611ac31b59f9 which would be one of the contributors to an automatic merge.

**Comment by JasonFengJ9** [comment]
Following https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/498#issuecomment-832870265, this draft PR doesn't have those very early commits.

**Comment by keithc-ca** [comment]
This updates the repo from jdk8u292-b10 to jdk8u302-b01 and thus should include all the following commits. (I don't know why some of these recent additions have dates 6 or more years in the past.)
```
$ git log jdk8u292-b10..jdk8u302-b01
commit 62c70425e09a7299c7ca85694d95611ac31b59f9
Merge: c705235b4a6 fcb2154b7cd
Author: J9 Build <j9build@ca.ibm.com>
Date:   Wed May 5 00:40:53 2021 +0000

    Merge sub-repo 'jdk' tag 'jdk8u302-b01' into master

commit c705235b4a6d5aedb87cbaa99bd575a5a4947978
Merge: 62c6828cd3e 6615e981b0a
Author: J9 Build <j9build@ca.ibm.com>
Date:   Wed May 5 00:40:49 2021 +0000

    Merge sub-repo 'langtools' tag 'jdk8u302-b01' into master

commit 62c6828cd3e213e659513bb87f37b8c308a86af4
Merge: 28bc0e4cfd0 dca928c2837
Author: J9 Build <j9build@ca.ibm.com>
Date:   Wed May 5 00:40:47 2021 +0000

    Merge root-repo 'jdk8u' tag 'jdk8u302-b01' into master

commit fcb2154b7cde34ce5715b94afd2c2134ce631f8f
Author: andrew <unknown>
Date:   Tue May 4 05:52:35 2021 +0100

    8258419: RSA cipher buffer cleanup
    Reviewed-by: phh, sgehwolf

commit 6615e981b0ac60be11a2ea41c427ce59cb5f7013
Merge: ccadd75921f b72c18e4d13
Author: andrew <unknown>
Date:   Fri Apr 23 01:47:35 2021 +0100

    Merge

commit dca928c28373de6e9558f9de9346a60f71ae81e7
Merge: 71173cbc4df ef05b6a841f
Author: andrew <unknown>
Date:   Fri Apr 23 01:47:32 2021 +0100

    Merge

commit ccadd75921fdaafbc0ea3042251c3c0588ba1c07
Merge: 3e03bc401f5 0a7d3d9b377
Author: andrew <unknown>
Date:   Thu Apr 22 01:33:31 2021 +0100

    Merge

commit 71173cbc4df343b602d1d9fc9287432dc0f71781
Merge: 3a3c0f68a51 71bc65e82ad
Author: andrew <unknown>
Date:   Tue Apr 20 19:34:10 2021 +0100

    Merge

commit 71bc65e82ad05fd23c1afdd89db64f663f20cc65
Author: aleonard <unknown>
Date:   Thu Apr 8 15:42:48 2021 +0000

    8264509: jdk8u MacOS zipped debug symbols won't build
    Summary: Copy diz file
    Reviewed-by: phh

commit 3e03bc401f58f6aa44447464061455a87b9b0218
Merge: 699a09009c1 b72c18e4d13
Author: andrew <unknown>
Date:   Wed Mar 17 20:42:12 2021 +0000

    Merge

commit 3a3c0f68a51a6c7e81a51c502d67793bc159195c
Merge: 4eb17a803be ef05b6a841f
Author: andrew <unknown>
Date:   Wed Mar 17 20:41:30 2021 +0000

    Merge

commit 699a09009c1b233890ee79cfd7434f19f13513ff
Author: bsrbnd <unknown>
Date:   Thu Mar 11 11:04:31 2021 +0800

    8214345: infinite recursion while checking super class
    Reviewed-by: vromero

commit 4eb17a803be3726bec8a25ad37fd219ec2bfed93
Author: aleonard <unknown>
Date:   Mon Mar 8 06:02:06 2021 +0000

    8262730: Enable jdk8u MacOS external debug symbols
    Reviewed-by: phh, andrew

commit 8522a9b0c2812d4975485c1069a6596acc898d1d
Author: jjiang <unknown>
Date:   Thu Jun 21 10:15:41 2018 +0800

    8205014: com/sun/jndi/ldap/DeadSSLLdapTimeoutTest.java failed with "Read timed out"
    Reviewed-by: sgehwolf

commit c226e4079b36c829a5e9950b14979e5422ec7f9f
Author: serb <unknown>
Date:   Thu Jul 6 15:54:39 2017 -0700

    8178403: DirectAudio in JavaSound may hang and leak
    Reviewed-by: prr, alitvinov

commit 47ec661785d16a75cd755460c10f19fa5816045c
Author: serb <unknown>
Date:   Wed Jun 26 15:34:13 2019 -0700

    8134672: [TEST_BUG] Some tests should check isDisplayChangeSupported
    Reviewed-by: prr

commit 5b8817becb2b9196acd451315ca9b2e05c2e2aa3
Author: shade <unknown>
Date:   Tue Apr 27 18:25:21 2021 +0200

    8265988: Fix sun/text/IntHashtable/Bug4170614 for JDK 8u
    Reviewed-by: sgehwolf

commit cbd1866f64b33994126c888df0ffae053a31b3cf
Author: jdv <unknown>
Date:   Wed Jul 13 17:34:26 2016 +0530

    7059970: Test case: javax/imageio/plugins/png/ITXtTest.java is not closing a file
    Reviewed-by: prr, bpb, psadhukhan
    Contributed-by: jayathirth.d.v@oracle.com

commit 06d13e6e8e85860207f1552e6a7549ffee13b609
Author: serb <unknown>
Date:   Sun Jan 3 05:08:48 2021 +0000

    8225116: Test OwnedWindowsLeak.java intermittently fails
    Reviewed-by: pbansal

commit 008a42efebfbf786ada73c0df48cf145201ffd1c
Author: serb <unknown>
Date:   Thu May 5 01:30:23 2016 +0300

    8151786: [TESTBUG] java/beans/XMLEncoder/Test4625418.java timed out intermittently
    Reviewed-by: alexsch, yan

commit 39637d109e866f1b2a35d231704c7b987edd144c
Author: serb <unknown>
Date:   Tue Aug 16 23:07:35 2016 +0300

    8159898: Negative array size in java/beans/Introspector/Test8027905.java
    Reviewed-by: alexsch, yan

commit b33af6b442bff4fc9a3d6932bb01e42905025c45
Author: rhalade <unknown>
Date:   Tue Feb 9 17:56:35 2021 +0000

    8225081: Remove Telia Company CA certificate expiring in April 2021
    Reviewed-by: mullan

commit 54aae2773d511be8c26ffd3e0334c5df84740c02
Author: xuelei <unknown>
Date:   Thu Mar 25 03:35:49 2021 +0000

    8258753: StartTlsResponse.close() hangs due to synchronization issues
    Reviewed-by: xuelei
    Contributed-by: Prajwal Kumaraswamy <pkumaraswamy@openjdk.org>

commit 5b81c930ba8ecad83695203319e8f25e956189eb
Author: bpb <unknown>
Date:   Thu May 18 12:53:19 2017 -0700

    8177809: File.lastModified() is losing milliseconds (always ends in 000)
    Summary: Use higher precision time values where available.
    Reviewed-by: bchristi, rriggs

commit 5e07d14a35e33e86a264696bcbded3e65486c288
Merge: bfb5c8dd891 dfe0e47f167
Author: andrew <unknown>
Date:   Fri Apr 23 01:47:38 2021 +0100

    Merge

commit bfb5c8dd8915f31eb72db991f8953829f75af21b
Author: naoto <unknown>
Date:   Fri Mar 26 17:13:49 2021 +0000

    8262110: DST starts from incorrect time in 2038
    8073446: TimeZone getOffset API does not  return a dst offset between years 2038-2137
    Reviewed-by: rriggs

commit fa84e220dd16ede7cd351079f14d7f4abc53ed4d
Author: arapte <unknown>
Date:   Thu Jul 16 06:35:01 2020 +0530

    8249278: Revert JDK-8226253 which breaks the spec of AccessibleState.SHOWING for JList
    Reviewed-by: prr, serb

commit ff5136a2b73018de6cdefa67ac19aea22eb0dfbb
Author: alexsch <unknown>
Date:   Wed Apr 22 15:00:50 2015 +0400

    8030123: java/beans/Introspector/Test8027648.java fails
    Reviewed-by: serb, alexsch
    Contributed-by: Nicholas Cull <run2000@gmail.com>

commit 09d371b5b843da16d383deeeb90283a9bb11fc53
Author: mbalao <unknown>
Date:   Mon Apr 12 20:58:08 2021 +0000

    8261355: No data buffering in SunPKCS11 Cipher encryption when the underlying mechanism has no padding
    Reviewed-by: valeriep, phh

commit 36b69eb6fe55bb1e1dcac1a15512bf8489c463c6
Author: xuelei <unknown>
Date:   Fri Apr 2 17:51:43 2021 +0800

    8259886: Improve SSL session cache performance and scalability
    Reviewed-by: erikj, xuelei, aph, mbalao
    Contributed-by: djelinski <djelinski1@gmail.com>

commit 02cf67d8bd50073018d1472be512d2c401816a19
Author: simonis <unknown>
Date:   Fri Nov 15 20:29:11 2019 +0100

    8234011: (zipfs) Memory leak in ZipFileSystem.releaseDeflater()
    Reviewed-by: clanger, lancea

commit 74efa246382556bb8d6a3592cb7af6cc123568dd
Author: andrew <unknown>
Date:   Wed Mar 31 22:03:54 2021 +0000

    8261867: Backport relevant test changes & additions from JDK-8130125
    Summary: Backport changes to bug6980209.java, JSplitPaneOverlapping.java, TestXEmbedServer.java, plus 3 new tests.
    Reviewed-by: phh, serb

commit 69aab278120c2a4df1a046cca2a1d33f6d95be46
Author: yan <unknown>
Date:   Tue Oct 20 12:42:21 2015 +0300

    8136592: [TEST_BUG] Fix 2 platform-specific closed regtests for jigsaw
    Reviewed-by: serb, yan
    Contributed-by: Renjith Alexander <renjith.alexander@oracle.com>

commit a270347d15b2f865b3da0041037a64f7d2e787c4
Author: aleonard <unknown>
Date:   Tue Mar 2 15:40:02 2021 +0000

    8262864: No debug symbols in image for Windows --with-native-debug-symbols=external
    Summary: Fix Images.gmk to keep debuginfo files
    Reviewed-by: sgehwolf, stooke

commit d2da04903d0284b2a138d5eb6a9e9c0670b4b9a2
Author: iignatyev <unknown>
Date:   Wed Jun 27 13:43:52 2018 -0700

    8199265: java/util/Arrays/TimSortStackSize2.java fails with OOM
    Reviewed-by: dholmes
    Contributed-by: Adam Farley <adfarley@redhat.com>

commit d939ee7559bb4bb79fbe19d98f9fdcd89d48b5ad
Author: iignatyev <unknown>
Date:   Tue Feb 27 21:29:08 2018 -0800

    8190679: java/util/Arrays/TimSortStackSize2.java fails with "Initial heap size set to a larger value than the maximum heap size"
    Reviewed-by: dholmes

commit fc1b72624fcb3dbd386afc9dfe5be7a41c915456
Author: lpriima <unknown>
Date:   Tue Mar 24 03:46:57 2015 -0400

    8075071: [TEST_BUG] TimSortStackSize2.java: OOME: Java heap space: MaxHeap shrinked by MaxRAMFraction
    Reviewed-by: dholmes
    Contributed-by: Adam Farley <adfarley@redhat.com>

commit 0a7d3d9b3777a01de15122c4836dc60135619841
Author: jjg <unknown>
Date:   Thu Dec 11 18:20:00 2014 -0800

    8066807: langtools/test/Makefile should use -agentvm not -samevm
    Reviewed-by: mcimadamore

commit 555ec935c6d7d21c8d96bb1111fc57e4dcc196e6
Author: shade <unknown>
Date:   Tue Apr 20 15:06:50 2021 +0200

    8260484: CheckExamples.java / NoJavaLangTest.java fail with jtreg 4.2
    Reviewed-by: sgehwolf

commit 9293f899115e104ae1b7fb8e87504fc61c9a7f74
Author: smarks <unknown>
Date:   Thu Mar 6 14:26:26 2014 -0800

    8036095: RMI tests using testlibrary.RMID and testlibrary.JavaVM do not pass through vmoptions
    Reviewed-by: alanb

commit 354b83855adaf2fe10dc9fcbc1e51b55ce8ca36b
Author: smarks <unknown>
Date:   Wed Feb 26 18:22:56 2014 -0800

    8263600: change rmidRunning to a simple lookup
    Reviewed-by: darcy

commit 3891f2e550cabf3e06fbac1684597e1fe177d31e
Author: mullan <unknown>
Date:   Tue Nov 24 18:14:05 2020 +0000

    8243559: Remove root certificates with 1024-bit keys
    Reviewed-by: weijun

commit ccb1b1b713eb95d7ff508298e40f0cde3a9a54f7
Author: jdv <unknown>
Date:   Mon Nov 20 11:02:54 2017 +0530

    8190332: PngReader throws NegativeArraySizeException/OOM error when IHDR width is very large
    Reviewed-by: prr, pnarayanan

commit 457d85350bd54c8bd085f453d912f16d32d44a0b
Merge: 18582087a5f 7604884e496
Author: andrew <unknown>
Date:   Wed Mar 17 20:41:45 2021 +0000

    Merge

commit 18582087a5f216c8e7883199aa5b40cc4613e644
Author: rhalade <unknown>
Date:   Thu Feb 4 19:11:57 2021 +0000

    8256421: Add 2 HARICA roots to cacerts truststore
    Reviewed-by: mullan

commit 152a4dcf68c023707d47befafd1479c5f8b562fc
Author: dfuchs <unknown>
Date:   Thu Oct 31 19:31:03 2019 +0000

    8231631: sun/net/ftp/FtpURLConnectionLeak.java fails intermittently with NPE
    Summary: sun/net/www/ftptest/FtpCommandHandler.java is modified to handle EOF properly
    Reviewed-by: chegar, vtewari

commit 479896759d03994d5528ea1b6b3a438a756589c5
Author: hshi <unknown>
Date:   Sat Mar 6 18:32:30 2021 +0800

    8263061: copy wrong unpack200 debuginfo to bin directory after 8252395
    Reviewed-by: sgehwolf

commit 0f102ab13d064df7387cd373c4409a53851bd016
Author: redestad <unknown>
Date:   Thu Mar 26 10:02:57 2020 +0100

    8241649: Optimize Character.toString
    Reviewed-by: redestad, rriggs
    Contributed-by: sergei.tsypanov@yandex.ru

commit 429dbc9cbada4fe12be827f8ec3ffca74af345a9
Author: sgehwolf <unknown>
Date:   Mon Mar 8 06:06:11 2021 +0000

    8255086: Update the root locale display names
    Reviewed-by: phh, andrew

commit 0e6a854f81d3f064e00b11993020cc2c9d74ca01
Author: sgehwolf <unknown>
Date:   Wed Mar 3 10:55:56 2021 +0100

    8180478: tools/launcher/MultipleJRE.sh fails on Windows because of extra-''
    Reviewed-by: sgehwolf
    Contributed-by: Masanori Yano <yano-masanori@fujitsu.com>

commit f8ffb897625590f6a6aa263787c1da5ae3b8d18e
Merge: 35da7a8e29e eb1d4ace145
Author: andrew <unknown>
Date:   Tue Mar 2 19:12:15 2021 +0000

    Merge

commit eb1d4ace1456081b84aa1664b406800d94e44910
Author: dongbohe <unknown>
Date:   Tue Feb 2 09:45:39 2021 +0800

    6878250: (so) IllegalBlockingModeException thrown when reading from a closed SocketChannel's InputStream
    Reviewed-by: alanb, andrew
    Contributed-by: songyaofei2@huawei.com
```

**Comment by keithc-ca** [comment]
As an example, the oldest newly included commit is:
```
commit 354b83855adaf2fe10dc9fcbc1e51b55ce8ca36b
Author: smarks <unknown>
Date:   Wed Feb 26 18:22:56 2014 -0800

    8263600: change rmidRunning to a simple lookup
    Reviewed-by: darcy
```
which is in the history of the new tag:
```
$ git merge-base --is-ancestor 354b83855adaf2fe10dc9fcbc1e51b55ce8ca36b jdk8u302-b01 && echo yes
yes
```

**Comment by JasonFengJ9** [comment]
Any theory why https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/498#issuecomment-833020389 doesn't include those aged commits?

**Comment by keithc-ca** [comment]
> Any theory why [#498 (comment)](https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/498#issuecomment-833020389) doesn't include those aged commits?

It looks like not all of the children of 35da7a8e29e67fb7d74da94742b08344ab9d34f8 were merged previously, but are now included in jdk8u302-b01.

In any case, I suggest we do want, for example, changes to `jdk/test/java/rmi/testlibrary/ActivationLibrary.java` to be attributed to '8263600: change rmidRunning to a simple lookup' (354b83855adaf2fe10dc9fcbc1e51b55ce8ca36b) rather than e372f6d6932a752e631185f7a2e18b2deb1a07f5 (a merge commit not from openjdk).

**Comment by JasonFengJ9** [comment]
> It looks like not all of the children of 35da7a8 were merged previously, but are now included in jdk8u302-b01.

Run the script again with latest updates, I see those commits as well.

> In any case, I suggest we do want, for example, changes to jdk/test/java/rmi/testlibrary/ActivationLibrary.java to be attributed to '8263600: change rmidRunning to a simple lookup' (354b838) rather than e372f6d (a merge commit not from openjdk).

If we take this path, the comments within `merge_master_to_staging.sh` need to be updated accordingly.

**Comment by keithc-ca** [comment]
I do plan to update the merge scripts.

**Review by JasonFengJ9 — APPROVED** [review]
LGTM

**Pull request merged**
